finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala

```diff
@@ -1,11 +1,36 @@
 package com.twitter.finagle.dispatch
 
 import com.twitter.finagle.context.Contexts
+import com.twitter.finagle.tracing.{Annotation, Trace}
 import com.twitter.finagle.transport.Transport
 import com.twitter.finagle.{Service, NoStacktrace, CancelledRequestException}
 import com.twitter.util._
 import java.util.concurrent.atomic.AtomicReference
 
+/**
+  * A base Server Dipatcher type that allows us to safely Annotate a
+  * dispatcher via
+  * [[com.twitter.finagle.dispatcher.ServerDispatherAnnotator ServerDispatcherAnnotator]]
+ */
+trait ServerDispatcher[Req, Rep, In, Out] {
+
+  /**
+   * Dispatches a request. The first argument is the request. The second
+   * argument `eos` (end-of-stream promise) must be fulfilled when the request
+   * is complete.
+   *
+   * For non-streaming requests, `eos.setDone()` should be called immediately,
+   * since the entire request is present. For streaming requests,
+   * `eos.setDone()` must be called at the end of stream (in HTTP, this is on
+   * receipt of last chunk). Refer to the implementation in
+   * [[com.twitter.finagle.http.codec.HttpServerDispatcher]].
+   */
+  private[dispatch] def dispatch(req: Out, eos: Promise[Unit]): Future[Rep]
```

Comment:
We can't make it `private[dispatch]` because we have `ServerDispatchers` outside of the `dispatch` package.  Let's make it public and call it a day–I don't think scala has the ability to say, "let subclasses fiddle with this method on other instances of this type".